### DISCUSSION TITLE
conditional statement para mostrar um container

### DISCUSSION THREAD

**Discussion by Dablinlabs** [Flutterando/forum#20] [Perguntas e Respostas] [answered]
Opa salve galera to começando agora, estou precisando de uma ajuda.
To com uma duvida em como usar um condicional para mostrar containers diferentes
tipo: 

if (selectedButton == 1) {
    //Container 1
} else if (selectedButton == 2) {
    //Container 2
}
.
.
.

**Comment by Dablinlabs** [comment]
opa consegui aqui ja, mas se alguem conhecer uma forma melhor

if (selectedButton == 1) ...[
//Container 1
] else if (selectedButton == 2)...[
//Container 2
]

**Comment by eduardoflorence** [comment]
A maneira que acho mais prática e legível é usando um Column ou Row, pois como os dois precisam de uma lista e o dart possui um recurso de item condicional nas listas, você pode fazer assim:
```dart
Column(children: [
  if (selectedButton == 1) Container(), // 1
  if (selectedButton == 2) Container(), // 2
])
```
Você também pode fazer de forma prática com o operador ternário, mas a legibilidade no seu cenário ficaria muito ruim.

**Comment by matheusrmribeiro** [comment]


**Reply by tiagualvs** [reply]
Criando uma função fica bem mais organizado, isso se você tiver varios tipos de containers para retornar como no exemplo aqui!

**Comment by eduardoflorence** [comment, marked as answer]
@matheusrmribeiro e @tihrasguinho, a equipe do Flutter já orientou que não se deve utilizar métodos para produzir Widgets (eles chamam de Helper Method), pois a performance fica ruim. Eles explicam isto neste vídeo: https://www.youtube.com/watch?v=IOyq-eTRhvo.
@matheusrmribeiro concordo com você que, dependendo da complexidade, a melhor solução seria isolar a parte variável, mas não com um método e sim construindo um Widget novo, que recebe como parâmetro o valor de selectedButton e, no seu método build, verifica o que deve ser mostrado.

**Reply by GabrielCR99** [reply]
Complementando esse comentário, retornar Widgets em métodos já é comprovado que perde a performance, como visto até no vídeo. Para quem quiser também colocar como uma rule no projeto Dart/Flutter (e que use o Dart Code Metrics como um plugin do analyzer), é essa:

https://dartcodemetrics.dev/docs/rules/flutter/avoid-returning-widgets

**Reply by tauamendonca** [reply]
@Lanysc essa resposta (incluindo a parte de isolar a parte variável e construir um widget novo) junto com a [outra](https://github.com/Flutterando/forum/discussions/20#discussioncomment-3636420)  apresentada pelo Eduardo Florence atendem o que você perguntou sobre melhores formas de colocar essa seleção. 
Movi sua pergunta para a categoria de "Perguntas e Respostas" para ficar melhor categorizado.
